tests/basecli_test.go

```diff
@@ -0,0 +1,222 @@
+package main
+
+import (
+	"bufio"
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	//"strings"
+	"testing"
+	"time"
+)
+
+// Tests assume the `basecoind` and `basecli` binaries
+// have been built and are located in `./build`
+
+// TODO remove test dirs if tests are successful
+
+var (
+	basecoind = "build/basecoind"
+	basecli   = "build/basecli"
+
+	basecoindDir = "./tmp-basecoind-tests"
+	basecliDir   = "./tmp-basecli-tests"
+
+	ACCOUNTS = []string{"alice", "bob", "charlie", "igor"}
+	alice    = ACCOUNTS[0]
+	bob      = ACCOUNTS[1]
+	charlie  = ACCOUNTS[2]
+	igor     = ACCOUNTS[3]
+)
+
+func gopath() string {
+	return filepath.Join(os.Getenv("GOPATH"), "src", "github.com", "cosmos", "cosmos-sdk")
+}
+
+func whereIsBasecoind() string {
+	return filepath.Join(gopath(), basecoind)
+}
+
+func whereIsBasecli() string {
+	return filepath.Join(gopath(), basecli)
+}
+
+func TestInitBaseCoin(t *testing.T) {
+	clean()
+
+	var err error
+
+	password := "some-random-password"
+	usePassword := exec.Command("echo", password)
+
+	initBasecoind := exec.Command(whereIsBasecoind(), "init", "--home", basecoindDir)
+
+	initBasecoind.Stdin, err = usePassword.StdoutPipe()
+	if err != nil {
+		t.Error(err)
+	}
+
+	initBasecoind.Stdout = os.Stdout
+
+	if err := initBasecoind.Start(); err != nil {
+		t.Error(err)
+	}
+	if err := usePassword.Run(); err != nil {
+		t.Error(err)
+	}
+	if err := initBasecoind.Wait(); err != nil {
+		t.Error(err)
+	}
+
+	if err := makeKeys(); err != nil {
+		t.Error(err)
+	}
+}
+
+func makeKeys() error {
+	var err error
```

Comment:
compress code, bring err into function definition: `func makeKeys() (err error) {`

Reply:
actually based on next comments, We shouldn't be even returning an error, but should pass in `t` anf use assert:
`func makeKeys(t *test.Testing) {`